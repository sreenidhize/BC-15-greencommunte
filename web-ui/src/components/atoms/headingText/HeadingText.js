import React from "react";
import PropTypes from "prop-types";
import { makeStyles, Typography } from "@material-ui/core";

const useStyles = makeStyles(() => ({
  headingDiv: {
    maxWidth: "500px",
  },
}));

const HeadingText = ({ label }) => {
  const classes = useStyles();
  return (
    <div
      id="headingText"
      className={classes.headingDiv}
      data-testid="heading-text"
    >
      <Typography variant={"h6"}>{label}</Typography>
    </div>
  );
};

HeadingText.propTypes = {
  label: PropTypes.string.isRequired,
};

export default HeadingText;
