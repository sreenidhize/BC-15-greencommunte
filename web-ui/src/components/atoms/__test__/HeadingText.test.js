import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import HeadingText from "../headingText/HeadingText";

describe("Heading text", () => {
    it("Matches the snapshot", () => {
        const { container } = render(
            <HeadingText label={"Heading"}/>
        );
        expect(container).toMatchSnapshot();
    });

    test("test heading render", () => {
        const { getByTestId } = render(<HeadingText label={"Heading"}/>);
        expect(getByTestId("heading-text")).toBeInTheDocument();
    });
});