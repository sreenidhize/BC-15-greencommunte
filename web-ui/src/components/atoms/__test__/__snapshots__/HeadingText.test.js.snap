// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading text Matches the snapshot 1`] = `
<div>
  <div
    class="makeStyles-headingDiv-1"
    data-testid="heading-text"
    id="headingText"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6"
    >
      Heading
    </h6>
  </div>
</div>
`;
